### PR TITLE
Fixed Bugs

### PR DESCRIPTION
**Pull request by DikshantK2004**
Issue: #12

### Short description of what this resolves:
- Passing the jwt token upon the successful login.
- jwt token is verified properly now, as it split along '.' and then decoded to get the user.
- in Room Creation, the status of isCreated = true and room Creation is done as transaction.
-  Updated Readme with the new files(folder structure)
-  Using JWT_SECRET using env
- Minor changes in API_ENDPOINTS.md and env.example

### Screenshots
image
image

### CODE REVIEW

File: src/controllers/userAuth.controllers.js
Comment (bhupesh98) [inline src/controllers/userAuth.controllers.js:1]: Token should also be given on register

File: src/middlewares/userAuth.middleware.js [hunk 1]
```diff
@@ -2,16 +2,15 @@ import jwt from "jsonwebtoken";
 
 async function fetchUser(req, res, next) {
     const authHeader = req.headers.authorization;
-    const token = authHeader && authHeader.split(" ")[1];
-
+    const token = authHeader && authHeader.split(".")[1];
```
Comment (bhupesh98) [inline src/middlewares/userAuth.middleware.js]: Bearer part is only removed everything else is the token

File: src/controllers/room.controllers.js
Comment (bhupesh98) [inline src/controllers/room.controllers.js:1]: jwt token upgrade was also expected

**Comment by DikshantK2004** [comment]
I also fixed checking if a room exists to add the question(when I made the commit question controller was not there)
. Forgot to mention this in update message

**Comment by bhupesh98** [comment]
While the issue is competitive, so after finalising we are giving all those participants partial points which will be manually assigned to you as a token of appreciation.

Closing this PR as we already accepted the closest one.

**Pull request closed without merging**